### PR TITLE
HEEDLS-778 : Display all centre courses on delegate courses

### PR DESCRIPTION
**Pull request by GeorgeBlackwell**
### JIRA link
_HEEDLS-778_

### Description
_Only centre specific courses were being shown on Delegate Courses when we should be showing all centre courses as well._

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [X] Run the formatter and made sure there are no IDE errors.
- [X] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [X] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [X] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/Services/CourseService.cs [hunk 1]
```diff
@@ -146,6 +154,22 @@ public IEnumerable<CourseStatistics> GetTopCourseStatistics(int centreId, int? c
             );
         }
 
+        public IEnumerable<CourseStatisticsWithAdminFieldResponseCounts>
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data/Services/CourseService.cs:157]: Is this only public for unit testing?

File: DigitalLearningSolutions.Data/Services/CourseService.cs [hunk 2]
```diff
@@ -63,6 +69,8 @@ RemovalMethod removalMethod
 
         public CentreCourseDetails GetCentreCourseDetails(int centreId, int? categoryId);
 
+        public CentreCourseDetails GetDelegateCourseDetails(int centreId, int? categoryId);
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data/Services/CourseService.cs:72]: I don't think the naming is very good for this method. It implies to me that it has something to do with getting a course for a delegate, but it doesn't.

**Review by AlexJacksonDS — CHANGES_REQUESTED** [review]
Mostly looks fine, just a naming issue. Also some merge conflicts to resolve

File: DigitalLearningSolutions.Data/Services/CourseService.cs (hunk 1 above)
Comment (GeorgeBlackwell) [reply DigitalLearningSolutions.Data/Services/CourseService.cs:157]: Yep 🙂

**Review by JBsoftwire — CHANGES_REQUESTED** [review]
Just a comment about method ordering in the file.

File: DigitalLearningSolutions.Data/Services/CourseService.cs [hunk 3]
```diff
@@ -244,6 +268,17 @@ public IEnumerable<CourseStatistics> GetTopCourseStatistics(int centreId, int? c
             return orderedCourses.Select(c => (c.CustomisationId, c.CourseName));
         }
 
+        public CentreCourseDetails GetDelegateCourseDetails(int centreId, int? categoryId)
```
[the remark this answers is not in the document]
Comment (GeorgeBlackwell) [reply DigitalLearningSolutions.Data/Services/CourseService.cs:271]: Yeah, Rider's auto implementer doesn't always order things as you'd expect 😵

**Review by JBsoftwire — APPROVED** [review]
LGTM

**Pull request merged**